src/types/Subject/Subject.ts

```diff
@@ -0,0 +1,8 @@
+import { TypePromise } from 'types/TypePromise/TypePromise';
+
+export type Subject = {
+  subject_id: string;
+  subject_name: string;
+};
+
+export interface SubjectPromise extends TypePromise<Subject[]> {}
```

Comment:
I admire how you are handling Promises from backend. I like the structure a lot.